test/parallel/test-crypto-authenticated.js

```diff
@@ -497,3 +698,159 @@ for (const test of TEST_CASES) {
+                            'qkuZpJWCewa6S',
+                            {
+                              authTagLength
+                            });
+    }, {
+      type: TypeError,
+      code: 'ERR_INVALID_OPT_VALUE',
+      message: `The value "${authTagLength}" is invalid for option ` +
+               '"authTagLength"'
+    });
+
+    common.expectsError(() => {
+      crypto.createDecipheriv('aes-256-ccm',
+                              'FxLKsqdmv0E9xrQhp0b1ZgI0K7JFZJM8',
+                              'qkuZpJWCewa6S',
+                              {
+                                authTagLength
+                              });
+    }, {
+      type: TypeError,
+      code: 'ERR_INVALID_OPT_VALUE',
+      message: `The value "${authTagLength}" is invalid for option ` +
+               '"authTagLength"'
+    });
+
+    if (!common.hasFipsCrypto) {
+      common.expectsError(() => {
+        crypto.createCipher('aes-256-ccm', 'bad password', { authTagLength });
+      }, {
+        type: TypeError,
+        code: 'ERR_INVALID_OPT_VALUE',
+        message: `The value "${authTagLength}" is invalid for option ` +
+                 '"authTagLength"'
+      });
+
+      common.expectsError(() => {
+        crypto.createDecipher('aes-256-ccm', 'bad password', { authTagLength });
+      }, {
+        type: TypeError,
+        code: 'ERR_INVALID_OPT_VALUE',
+        message: `The value "${authTagLength}" is invalid for option ` +
+                 '"authTagLength"'
+      });
+    }
+  }
+}
+
+// Test that create(De|C)ipher(iv)? throws if the mode is CCM and no
+// authentication tag has been specified.
+{
+  assert.throws(() => {
+    crypto.createCipheriv('aes-256-ccm',
+                          'FxLKsqdmv0E9xrQhp0b1ZgI0K7JFZJM8',
+                          'qkuZpJWCewa6S');
+  }, /^Error: authTagLength required for aes-256-ccm$/);
+
+  assert.throws(() => {
+    crypto.createDecipheriv('aes-256-ccm',
+                            'FxLKsqdmv0E9xrQhp0b1ZgI0K7JFZJM8',
+                            'qkuZpJWCewa6S');
+  }, /^Error: authTagLength required for aes-256-ccm$/);
+
+  if (!common.hasFipsCrypto) {
+    assert.throws(() => {
+      crypto.createCipher('aes-256-ccm', 'very bad password');
+    }, /^Error: authTagLength required for aes-256-ccm$/);
+
+    assert.throws(() => {
+      crypto.createDecipher('aes-256-ccm', 'very bad password');
+    }, /^Error: authTagLength required for aes-256-ccm$/);
+  }
+}
+
+// Test that setAAD throws if an invalid plaintext length has been specified.
+{
+  const cipher = crypto.createCipheriv('aes-256-ccm',
+                                       'FxLKsqdmv0E9xrQhp0b1ZgI0K7JFZJM8',
+                                       'qkuZpJWCewa6S',
+                                       {
+                                         authTagLength: 10
+                                       });
+
+  for (const plaintextLength of [-1, true, false, NaN, 5.5]) {
```

Comment:
ditto for `0` in here

Reply:
Zero is a valid value meaning that the message is empty, but I can add a test case if you would like.